### ISSUE TITLE
Field type array cannot handle nested type row

### ISSUE THREAD

**Issue by Cuzart** [pemedia/payload-visual-editor#4]
Hey there,
first of all thanks for the great contribution with this plugin. When I tried it out I have recognized that the entries of my Faq Block are not populated correctly. I have defined the following template for that block 

```ts
 {
      name: "entries",
      type: "array",
      labels: ...,
      fields: [
        {
          type: "row", // this is the culprit
          fields: [
            {
              name: "question",
              label: { en: "Question", de: "Frage" },
              type: "text",
            },
            {
              name: "answer",
              label: { en: "Answer", de: "Antwort" },
              required: true,
              type: "richText",
            },
          ],
        },
      ],
    },
```

and for my page data I am receiving `0: {undefined: undefined, id: '6463c80500acb23cf74cbf31'}` instead of the populated question and answer properties.

I checked your code and I have seen that you are handling field type array in _generateDocument.ts_ and used some console logs to better understand why this happens: 
In the docs the row field is defined as the following: "The Row field is presentational-only and only affects the Admin panel. By using it, you can arrange fields next to each other horizontally." that is why it does not affect the JSON structure coming from the REST API. As you are getting the field data from the admin panel the structure is changed 
<img width="1157" alt="Bildschirmfoto 2023-06-16 um 11 52 39" src="https://github.com/pemedia/payload-visual-editor/assets/60718229/59175c18-9d23-494d-b7cb-3c06f020e46b">

If wanted I could support by creating a PR.

**Comment by dkirchhof** [comment]
Hey, thanks for your feedback. Presentational only fields should be flattened. So the `getAllFields` function could be extended for it. If you wan't to create a PR, feel free. Otherwise I could fix it in the next days.

**Comment by dkirchhof** [comment]
@Cuzart would you like to check out, if the PR fixes your issue?

**Comment by Cuzart** [comment]
@dkirchhof  Yep that did the trick. You were faster than me 😄 but I like your solution thanks for the quick response.
